lib/page/surveydetail/survey_detail_page.dart

```diff
@@ -66,16 +141,37 @@ class SurveyDetailPage extends StatelessWidget {
                       buttonText:
                           AppLocalizations.of(context)!.surveyDetailStartSurvey,
                       onPressed: () {
-                        // TODO: Navigate to Questions screen
+                        if (surveyDetail != null) {
+                          _navigateToQuestion(surveyDetail);
+                        } else {
+                          if (survey != null)
+                            ref
+                                .read(surveyDetailViewModelProvider.notifier)
+                                .getSurveyDetail(survey, isRetry: true);
+                        }
                       },
                     ),
                   ),
                 ],
               ),
             ),
           ),
+          if (shouldShowLoading)
+            LoadingIndicatorWidget(shouldIgnoreOtherGestures: true),
         ],
       ),
     );
   }
+
+  void _navigateToQuestion(SurveyDetailModel surveyDetail) =>
+      _appNavigator.navigateToQuestion(context, surveyDetail);
+
+  void _showError(String errorMessage) {
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+        duration: const Duration(seconds: Constants.snackBarDurationInSecond),
+        content: Text(errorMessage),
+      ));
+    });
```

Comment:
we don't always need to wrap the snackbar showing inside of `WidgetsBinding.instance.addPostFrameCallback`. Please re-check 🙏

Reply:
@chornerman okay, it seems we still need it this case.